tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerHelper.java

```diff
@@ -53,176 +49,102 @@ public final class TinkerHelper {
         return null != graph.graphComputerView;
     }
 
-    public static TinkerGraphComputerView createGraphComputerView(final TinkerGraph graph, final GraphFilter graphFilter, final Set<VertexComputeKey> computeKeys) {
+    public static TinkerGraphComputerView createGraphComputerView(final AbstractTinkerGraph graph, final GraphFilter graphFilter, final Set<VertexComputeKey> computeKeys) {
         return graph.graphComputerView = new TinkerGraphComputerView(graph, graphFilter, computeKeys);
     }
 
-    public static TinkerGraphComputerView getGraphComputerView(final TinkerGraph graph) {
+    public static TinkerGraphComputerView getGraphComputerView(final AbstractTinkerGraph graph) {
         return graph.graphComputerView;
     }
 
-    public static void dropGraphComputerView(final TinkerGraph graph) {
+    public static void dropGraphComputerView(final AbstractTinkerGraph graph) {
         graph.graphComputerView = null;
     }
 
-    public static Map<String, List<VertexProperty>> getProperties(final TinkerVertex vertex) {
-        return null == vertex.properties ? Collections.emptyMap() : vertex.properties;
-    }
-
-    public static void autoUpdateIndex(final TinkerEdge edge, final String key, final Object newValue, final Object oldValue) {
-        final TinkerGraph graph = (TinkerGraph) edge.graph();
-        if (graph.edgeIndex != null)
-            graph.edgeIndex.autoUpdate(key, newValue, oldValue, edge);
-    }
-
-    public static void autoUpdateIndex(final TinkerVertex vertex, final String key, final Object newValue, final Object oldValue) {
-        final TinkerGraph graph = (TinkerGraph) vertex.graph();
-        if (graph.vertexIndex != null)
-            graph.vertexIndex.autoUpdate(key, newValue, oldValue, vertex);
-    }
-
-    public static void removeElementIndex(final TinkerVertex vertex) {
-        final TinkerGraph graph = (TinkerGraph) vertex.graph();
-        if (graph.vertexIndex != null)
-            graph.vertexIndex.removeElement(vertex);
-    }
-
-    public static void removeElementIndex(final TinkerEdge edge) {
-        final TinkerGraph graph = (TinkerGraph) edge.graph();
-        if (graph.edgeIndex != null)
-            graph.edgeIndex.removeElement(edge);
-    }
-
-    public static void removeIndex(final TinkerVertex vertex, final String key, final Object value) {
-        final TinkerGraph graph = (TinkerGraph) vertex.graph();
-        if (graph.vertexIndex != null)
-            graph.vertexIndex.remove(key, value, vertex);
-    }
-
-    public static void removeIndex(final TinkerEdge edge, final String key, final Object value) {
-        final TinkerGraph graph = (TinkerGraph) edge.graph();
-        if (graph.edgeIndex != null)
-            graph.edgeIndex.remove(key, value, edge);
-    }
-
+    // todo: move to TinkerVertex?
     public static Iterator<TinkerEdge> getEdges(final TinkerVertex vertex, final Direction direction, final String... edgeLabels) {
```

Comment:
Why were these changes necessary?

Reply:
I changed the way how vertices contains linked edges. The main reason is the ability to set a transaction boundary.